app/qml/map/MMMapController.qml

```diff
@@ -35,6 +35,10 @@ Item {
   property bool isStreaming: recordingToolsLoader.active ? recordingToolsLoader.item.recordingMapTool.recordingType === MM.RecordingMapTool.StreamMode : false
   property bool centeredToGPS: false
 
+  property var mapToolComponent: {
+    state === "measure" ? measurementToolsLoader.item : null
```

Comment:
I'd recommend binding this as 
```suggestion
    measurementToolsLoader.active ? measurementToolsLoader.item : null
```

In this case you will be sure that the loader will first get active and only after that its item will be assigned here.

Reply:
I think we are only interested in the mapTool (the cpp class), aren't we? In that case, we should pass just `measurementToolsLoader.item.mapTool`

Reply:
For `finalizeMeasurement` and `repeatMeasure` methods inside `MMMeasurementTools`, we need to modify the mapTool, guidelineController and crosshair. Therefore, we need access to both `mapTool` and `MMMeasurementTools` instance.